app.js

```diff
@@ -10,8 +11,174 @@ class KNXApp extends Homey.App {
+        });
+    } catch (e) {
+      console.log(e);
+    }
+  }
+
+  /**
+   * Handler for the interface found
+   */
+  onKNXInterface(knxInterface) {
+    this.receiveTelegramTrigger.getArgumentValues().then(this.registerKNXEventHandlers.bind(this));
+  }
+
+  registerKNXEventHandlers(cards) {
+    cards.forEach((args) => {
+      if (!this.eventListenerGroupAddresses.includes(`${args.interface.mac}-${args.group_address}`)) {
+        this.log('adding event listener', `${args.interface.mac}-${args.group_address}`);
+
+        let knxInterfaceToUse = this.knxInterfaceManager.getKNXInterface(args.interface.mac);
+        if (args.interface.mac === 'any') {
+          const availableInterfaces = this.knxInterfaceManager.getKNXInterfaceList();
+          const macs = Object.keys(availableInterfaces);
+          knxInterfaceToUse = macs.length > 0 ? availableInterfaces[macs[0]] : null;
+        }
+
+        if (!knxInterfaceToUse) {
+          this.error('Interface not found');
+        } else {
+          this.eventListenerGroupAddresses.push(`${args.interface.mac}-${args.group_address}`);
+          // Store the event listener so we can remove it later
+          knxInterfaceToUse.addKNXEventListener(args.group_address, this.onKNXEvent.bind(this, args));
+        }
+      }
+    });
+
+    this.eventListenerGroupAddresses.forEach((eventListernerId) => {
+      if (cards.filter((args) => `${args.interface}-${args.group_address}` === eventListernerId).length === 0) {
+        // TODO: figuure out how to remove the event listener, because we need the correct event listener
+        // knxInterfaceToUse.removeKNXEventListener(groupAddress, this.onKNXEvent.bind( this, args));
+      }
+    });
+  }
+
+  async interfaceAutocomplete(query, args) {
+    const results = this.knxInterfaceManager.getSimpleInterfaceList();
+    results.push({ name: '[any]', mac: 'any', ip: 'any' });
+
+    // filter based on the query
+    return results.filter((result) => (
+      result.name.toLowerCase().includes(query.toLowerCase())
+      || result.mac.toLowerCase().includes(query.toLowerCase())
+      || result.ip.toLowerCase().includes(query.toLowerCase())
+    ));
+  }
+
+  async datatypeAutocomplete(query, args) {
+    return this.availableKnxDataTypes.filter((r) => r.name.includes(query.toUpperCase()));
+  }
+
+  async sendKNXTelegram(args, state) {
+    console.log(args);
+    console.log(state);
+
+    let knxInterfaceToUse = this.knxInterfaceManager.getKNXInterface(args.interface.mac);
+    if (args.interface.mac === 'any') {
+      const availableInterfaces = this.knxInterfaceManager.getKNXInterfaceList();
+      const macs = Object.keys(availableInterfaces);
+      knxInterfaceToUse = macs.length > 0 ? availableInterfaces[macs[0]] : null;
+    }
+
+    if (!knxInterfaceToUse) {
+      return Promise.reject(new Error('Interface not found'));
+    }
+
+    if (!args.group_address) {
+      return Promise.reject(new Error('No group address selected'));
```

Comment:
These errors are not thrown to the user

Reply:
What would a good approach then be?

Reply:
I have refactored a bit, how does it look now?